test/functional/bulk_tests.js

```diff
@@ -644,99 +644,51 @@ describe('Bulk', function() {
     }
   });
 
-  it('should correctly handle multiple unordered batch API', {
-    metadata: { requires: { topology: ['single', 'replicaset', 'ssl', 'heap', 'wiredtiger'] } },
-
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+  it('should correctly handle multiple unordered batch API', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient(configuration.writeConcernMax(), {
+      poolSize: 1
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
-        var col = db.collection('batch_write_unordered_ops_legacy_2');
+    client.connect((err, client) => {
+      const db = client.db(configuration.db);
+      const col = db.collection('batch_write_unordered_ops_legacy_2');
 
-        // Add unique index on b field causing all updates to fail
-        col.ensureIndex({ a: 1 }, { unique: true, sparse: false }, function(err) {
-          test.equal(err, null);
+      // Add unique index on b field causing all updates to fail
+      col.ensureIndex({ a: 1 }, { unique: true, sparse: false }, err => {
+        expect(err).to.not.exist;
 
-          // Initialize the unordered Batch
-          var batch = col.initializeUnorderedBulkOp({ useLegacyOps: true });
+        // Initialize the unordered Batch
+        const batch = col.initializeUnorderedBulkOp({ useLegacyOps: true });
 
-          // Add some operations to be executed in order
-          batch.insert({ b: 1, a: 1 });
-          batch
-            .find({ b: 2 })
-            .upsert()
-            .updateOne({ $set: { a: 1 } });
-          batch
-            .find({ b: 3 })
-            .upsert()
-            .updateOne({ $set: { a: 2 } });
-          batch
-            .find({ b: 2 })
-            .upsert()
-            .updateOne({ $set: { a: 1 } });
-          batch.insert({ b: 4, a: 3 });
-          batch.insert({ b: 5, a: 1 });
+        // Add some operations to be executed in order
+        batch.insert({ b: 1, a: 1 });
+        batch.insert({ b: 4, a: 3 });
```

Comment:
I think we can also get rid of this operation since it's the insertion of a duplicate key (`{ a: 1 }`) that is causing the write error below.